.github/workflows/backend.yml

```diff
@@ -0,0 +1,445 @@
+name: backend
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  files-changed:
+    name: detect what files changed
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check for backend file changes
+        uses: getsentry/paths-filter@v2
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
+
+  api-docs:
+    needs: files-changed
+    name: api docs test
```

Comment:
I have copy/pasted all the workflows I deleted. I added the line `needs: files-changed` and the key for each job (e.g. `test:` to `api-docs:`).

I also changed the conditional steps from: `steps` to `needs`. See below.